src/components/Articles/Articles.vue

```diff
@@ -5,22 +5,29 @@
       <label class="define-article-title">Articles</label>
     </div>
 
-    <div class="section-container" v-if="getBusinessInformation.hasRestrictions">
+    <div
+      class="section-container"
+      v-if="getBusinessInformation.hasRestrictions"
+      :class="{'invalid-section': invalidCompanyProvisions}">
       <company-provisions
         class="sub-section"
         :provisionsRemoved="getProvisionsRemoved"
         @companyProvisionsChanged="setProvisionsRemoved($event)"
         @haveChanges="emitHaveChanges($event)"
+        @onEditingCompanyProvisions="setEditingCompanyProvisions($event)"
       />
     </div>
 
-    <div class="section-container" :class="{'invalid-section': !getIsResolutionDatesValid}">
+    <div
+      class="section-container"
+      :class="{'invalid-section': invalidResolutionDates}">
       <resolution-dates
         :addedDates="getNewResolutionDates"
         :previousDates="getPreviousResolutionDates"
         :isEditMode="true"
         :hasRightsOrRestrictions="getHasRightsOrRestrictions"
         @addRemoveDate="setResolutionDates($event)"
+        @onAddingResolutionDate="setIsAddingResolutionDate($event)"
```

Comment:
Again, this is really an event telling us if the component is in editing mode, right? So I would use the same verb as above: `@isEditing="..."`

What I'm doing here is taking a step back and looking at things... the Articles component doesn't care about the details of Resolution Dates component. It only cares that Resolution Dates is in edit mode. So that's all I would keep track of. And if we can make the component event handers consistent then that will be helpful for someone to generally handle the code. This project is almost big enough now that I don't understand everything in it, so keep my "immediate concerns" in scope and not thinking about what sub-components do is very helpful.

Reply:
Yes, I understand what you're saying but I need to know if the subcomponent isEditing to set the invalid state for the resolution-dates component. editMode is another thing related to check if the component is "editable".

Reply:
isAdding ?

Reply:
yes, I changed onAddingResolutionDate to isEditing to keep it consistent to the other components.